### DISCUSSION TITLE
Further plans for project

### DISCUSSION THREAD

**Discussion by bolDeju** [alonswartz/notesium#94] [General]
I like the way you are creating this project, standards should be a base for further app modules.

I have a few questions:
Are you going to implement a parsing of files for certain logic objects, like tasks?
Will there be some kind of inline math (mathjax, latex-like)?
Will there be some kind of native save-file - link it support (for example a science article in pdf, that you annotated in note will be saved in some `attachments` directory?
Will there be some kind of highlight tool for any kind of documents?

I am asking those questions because I want to help with project development.
What is your roadmap for this project at this moment?

**Comment by alonswartz** [comment]
> I like the way you are creating this project, standards should be a base for further app modules.

;)

> Are you going to implement a parsing of files for certain logic objects, like tasks?

Most likely. It's been on my todo list for a while, especially since the addition of the calendar widget and deterministic periodic notes. I'm not entirely sure of what it will look like. Happy to discuss ideas.

> Will there be some kind of inline math (mathjax, latex-like)?

Not currently planned.

> Will there be some kind of native save-file - link it support (for example a science article in pdf, that you annotated in note will be saved in some attachments directory?

Almost definately! I was working on `assets` support before I got side-tracked with the recent keyboard-layout issues (DE, RU). I'll post more information about this soon - would love your feedback and ideas.

> Will there be some kind of highlight tool for any kind of documents?

Hmmm... not sure. Are you thinking about something minimal or more in the lines of [zotero](https://www.zotero.org/) ?

> I am asking those questions because I want to help with project development.

That's awesome! Any help would be greatly appreciated.

> What is your roadmap for this project at this moment?

I don't have a publicly accessible roadmap. I'm currently tracking everything in my own notes (managed by Notesium). I should probably make one.

**Comment by bolDeju** [comment]
Thanks for your reply.

## Objects

### Tasks

First of all I think it is very helpful that [markdown](https://www.markdownguide.org/basic-syntax/) (by default don't tell `- [ ]` is a standart tasks syntax) because:
 - it is not easy to use if you are not familiar with modal editing (in vim or helix just `r` to `x` between `[ ]`)
 - you should always watch that there is space between `[ ]`
 - it is not easy to understand if you are *doing* task or **done** by one char (this is for source editing)

#### Alternatives to `- [ ]`

1. [org-mode inline tasks](https://orgmode.org/worg/org-syntax.html#Inlinetasks)
  - Personally used in Logseq, easy to read, easy to modify. In my opinion, complex syntax for bare-hand writing.
  - Will not recomend to use as a standalone task without shortcut infrastructure around it.
  - Have multiple line syntax.
  - I haven't seen many mobile apps for this syntax.
2. [todo.txt](https://github.com/todotxt/todo.txt)
  - Easy to read, easy to parse. Easy to add personal completion state. 
  - Many dedicated apps (android, linux, windows, ios).
  - May be a little too long to read in plain-text editor.
  - There is a complete, stabilized [format](https://github.com/todotxt/todo.txt-cli/wiki/The-Todo.txt-Format) that is not changing very long.
3. [plain markdown **BOLD** status](https://plaintextproject.online/articles/2017/07/25/tasks.html) for task
  - Easy to write, read. Easy personalization of tasks.
4. [obsidian dataview tasks](https://publish.obsidian.md/tasks/Introduction)
  - I just don't think that tasks should be marked with any kind of emoji, because it is not always easy to write them from cli, so to use it with cli apps.
  - I love how it look when it is parsed. It is easy to read in this way.

#### My thoughts

I think that by default there should be todo.txt tasks. All other formats should be able to be implemented by some kind of side-parser.
  I don't completely know how it is better to implement but here is the idea. There is a binary that can parse specialized type of object (org-mode tasks) and output it in stdout (json-format, newline, ... idk)
  Then app just should merge those tasks in some kind of unified table with tags from where did it come. (Parse of all notes dir or just one file, that was specified)
  User that can write a simple rule, what task he want: urgent and important, importand and not on time... etc.

This way, I think may be implemented any kind of object parsing. But there should be a BASE what should be always parsed first.

As for me, I found out that I only need several tools for note-taking and productivity:
  - Google calendar (without tasks, they are bad in Gcal)
    - just for basic time management, not task planning, because task planning is always wrong
  - Basic notes app (for jotting lecture notes or my personal ideas)
    - here I have Projects, Areas, Resources and Archive
    - Projects and Resources sometimes merged.
    - Projects have most of tasks.
    - Project always have a lot of external links. External pdfs, in what I highlight something.
    - Archive is just a bunch of old files that is currently not useful in any area of life.
    - Areas sometimes have tasks too.
  - Tasks app where I sort tasks by four categories:
    1. urgent and important
    2. not on time and important  
    3. urgent and **not** important
    4. not on time and **not** important

As you can see there only four objects that I definetly use:
  1. Calendar date
  2. Task
  3. Any kind of external file
  4. Link (about it later)

I don't know what any user definetly need. Those are my needs. I'd like to see parsing of them in app by default.

### Links

I'had a lot of links to old sites in my old Logseq database. Now a lot of small cool sites are gone and info from them too.
I think there is only way to solve this problem - saving such info by yourself. Any site can change. Your notes should refer something that you meant when was writing.
I've searched a lot for easy to use tools with such ability, but found only several:
 - omnivore
 - zotero
 - logseq pdf referencing
But all of them lack of minimalisticity - i just need 3 formats to export and highlight into:
  - pdf (standart, will be always easy to read, not something like doc wich will not be readable in libreoffice and even newer word)
  - image (vector and rastor)
  - plain-text (may be basic markdown, not some fancy javascript based it implementation)

I just need ability to save anything and then just add my thoughts to any part of that info i saved.
Zotero is not easy to use and in my opinion, more specialized (academic).
Links (like object) should refer:
  - `date` when it was done
  - `place` - local file or just link to a site
  - `context` - relative place in whole `place` what you are talking about (optional)

Currently I am just thinking of how it will be implemented in general way, `Do one thing, and do it well`.
How it should work with external tools? I don't currently know. I think this should be external, but easy to implement in other tools.

However, basic `[]()` markdown link is a must-have in terms of fast and easy-to-use writing. My teacher always said: `Every tool should do one thing right—not everything poorly.`.

**Comment by aronvandepol** [comment]
I also think the setup you have going on here. 

I'm not sure how extensive you have an interested in developing things like tasks, but with the periodic notes and calendar addition, I can imagine how you can move towards a simple task setup, or provide some automatic querying of tasks. e.g: task with due date shows up on the periodic notes?

Additionally integration with things like Taskwarrior are not far-fetched but perhaps beyond the scope of what this project aims to be. Curious to hear your take on how you yourself viewed the addition of tasks and to-do's.

**Comment by alonswartz** [comment]
Apologies for the delayed response.

I don’t think there’s a one-size-fits-all approach here. Different apps implement task management in different ways with different features, but more importantly, different people have different needs and workflows—and it’s important to recognize that.

I tend to prefer the philosophy of *"Do one thing and do it well"* rather than adding half-baked features. That said, I do think there’s something interesting here, especially when it comes to integrating tasks with periodic notes and the calendar. I’m just not quite sure that’s the right direction yet. Or maybe we can do better?

This needs a little more thought, especially when it comes to due dates.

---  

Anyway, what I have found interesting is that Notesium works really well when treating tasks more like **projects** rather than **checklist** items. This is how I use Notesium to manage my work journal, and I’ll probably write a blog post on it at some point.

For some context, I use a combination of weekly notes, **labels** for state, and **notes** for projects/tasks. I’ll skip over how I use weekly notes for now.

I use labels to track state, usually just `@inprogress` and `@deferred` to keep things simple. But you could extend this with `@todo`, `@done`, `@important`, `@blocked`, etc. The `@` symbol isn’t required—I just use it to keep labels sorted at the top alphabetically and to distinguish them from other labels. 

Then, when creating a note, I just link to the relevant labels and update them as needed. That’s it.

With this setup, I can query, slice, and dice my tasks however I need using the finder. It’s simple, flexible, powerful, and works right now. The only real limitation is handling due dates, but for everything else, this approach has been extremely effective.
